SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-3x/tsWxGTUI_Py3x/ManPages/tsManPagesLibCLI/tsGistGetTerminalSize.man
Help on module tsGistGetTerminalSize:

NAME
    tsGistGetTerminalSize

FILE
    /cygdrive/d/WR/Technical_Preview-Draft/tsWxGTUI_PyVx/Python-2x/tsWxGTUI_Py2x/tsLibCLI/tsGistGetTerminalSize.py

DESCRIPTION
    tsGistGetTerminalSize.py - Module for acquiring the character
    size of the Python console window as tuple (width, height) on host
    operating systems (such as Linux, Mac OS X, Microsoft Windows and
    Unix).

FUNCTIONS
    get_terminal_size()
        getTerminalSize()
        - get width and height of console
        - works on linux,os x,windows,cygwin(windows)
        originally retrieved from:
        http://stackoverflow.com/questions/566746/
               how-to-get-console-window-width-in-python

DATA
    __authors__ = 'Richard S. Gordon'
    __copyright__ = 'Copyright (c) 2013 Richard S. Gordon.\n\t\tAll rights...
    __credits__ = '\n\n  Credits: \n\n\t  terminalsize (https://gist.git.....
    __date__ = '03/22/2015'
    __header__ = '\n\ntsGistGetTerminalSize, v1.0.0 (build 03/22/201... Ge...
    __license__ = 'GNU General Public License, Version 3, 29 June 2007'
    __line1__ = 'tsGistGetTerminalSize, v1.0.0 (build 03/22/2015)'
    __line2__ = 'Author(s): Richard S. Gordon'
    __line3__ = 'Copyright (c) 2013 Richard S. Gordon.\n\t\tAll rights res...
    __line4__ = 'GNU General Public License, Version 3, 29 June 2...U Gene...
    __title__ = 'tsGistGetTerminalSize'
    __version__ = '1.0.0'
    mainTitleVersionDate = 'tsGistGetTerminalSize, v1.0.0 (build 03/22/201...

VERSION
    1.0.0

DATE
    03/22/2015

CREDITS
    
    
      Credits: 
    
    	  terminalsize (https://gist.github.com/jtriley/1108174) Features: 
    	  Copyright (c) 2011 Justin T. Riley.
    			All rights reserved.
    	  GNU General Public License, Version 3, 29 June 2007




Documents/NEWS.txt
#-----------------------------------------------------------
#"Time-stamp: <07/08/2015  4:48:41 AM rsg>"
#-----------------------------------------------------------

========================= NEWS.txt =========================

   +----+----+  TeamSTARS "tsWxGTUI_PyVx" Toolkit
   | ts | Wx |      with Python 2x & Python 3x based
   +----+----+         Command Line Interface (CLI)
   | G T U I |      and "Curses"-based "wxPython"-style, 
   +---------+         Graphical-Text User Interface (GUI)
 
   Get that cross-platform, pixel-mode "wxPython" feeling
   on character-mode 8-/16-color (xterm-family) & non-color
   (vt100-family) terminals and terminal emulators.

   You can find this and other plain-text files in the
   Toolkit subdirectory named:

       "./<Toolkit Recipient's Repository>/Documents".

       <Your Working Repository>
       (e.g. "tsWxGTUI_PyVx_Repository") 
         |
         |
         +-- ["Documents"]

=========================== News ===========================

The pre-alpha release of the TeamSTARS "tsWxGTUI_PyVx-0.0.0"
Toolkit is now available. It includes the following two
versions abd their associated variants (Python Distribution
Utilities ("Distutils") installable "Site-Packages" and non-
installable "Developer-Sandboxes" for enhancement and bug
fix development and experimentation without interference
with the already installed Site-Packages):

1. The TeamSTARS "tsWxGTUI_Py2x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   second generation Python 2.x programming language which
   is widely available.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. The final 2.x
   version 2.7 release came out in mid-2010, with a state-
   ment of extended support for this end-of-life release.
   The 2.x branch will see no new major releases after that.

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 2.0.0-2.7.9.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 2.6.4-2.7.9.

2. The TeamSTARS "tsWxGTUI_Py3x-0.0.0" Toolkit is released
   for users of the popular, high-level, cross-platform,
   third generation Python 3.x programming language which
   is growing in availability.

   A wiki posting by the Python Software Foundation announc-
   ed that Python 3.0 was released in 2008. 3.x is under
   active development and has already seen over five years
   of stable releases, including version 3.3 in 2012 and
   3.4 in 2014. This means that all recent standard library
   improvements, for example, are only available by default
   in Python 3.x.

   Guido van Rossum (the original creator of the Python
   language) decided to clean up Python 2.x properly,
   with less regard for backwards compatibility than is
   the case for new releases in the 2.x range. The most
   drastic improvement is the better Unicode support
   (with all text strings being Unicode by default) as
   well as saner bytes/Unicode separation.

   Besides, several aspects of the core language (such as
   print and exec being statements, integers using floor
   division) have been adjusted to be easier for newcomers
   to learn and to be more consistent with the rest of the
   language, and old cruft has been removed (for example,
   all classes are now new-style, "range()" returns a memory
   efficient iterable, not a list as in 2.x).

   The source code and documentation is available for down-
   load at:

      http://pypi.python.org/pypi.

   It supports development and operation of:

   a. Command Line User Interface (CLI) application
      programs implemented in Python 3.0.0-3.4.2.

   b. Graphical-style User Interface (GUI) application
      programs implemented in Python 3.1.5-3.4.2.

======================= End-Of-File ========================


SourceDistributions/Developer-Sandboxes/tsWxGTUI_PyVx/Python-2x/tsWxGTUI_Py2x/tsUtilities/tsGistGetTerminalSize.py
#! /usr/bin/env python
#"Time-stamp: <10/17/2013  8:08:39 AM rsg>"
'''
tsGistGetTerminalSize.py - Module for acquiring the character
size of the Python console window as tuple (width, height) on host
operating systems (such as Linux, Mac OS X, Microsoft Windows and
Unix).
'''

#################################################################
#
# File: tsGistGetTerminalSize.py
#
# Purpose:
#
#     Module for acquiring the character size of the Python
#     console window as tuple (width, height) on host operating
#     systems (such as Linux, Mac OS X, Microsoft Windows and
#     Unix).
#
# Limitations:
#
#    None.
#
# Notes:
#
#    Derived from: https://gist.github.com/jtriley/1108174
#    by Justin T. Riley, created "terminalsize.py" for
#    gist 2011-07-26T14:59:00-07:00.
#
# Usage:
#
#    ## Import example:
#
#        import tsGistGetTerminalSize
#
#    ## Execute example:
#
#        python tsGistGetTerminalSize.py
#
# Methods:
#
#    __main__ - Print tuple (width, height) on linux, mac os x,
#                          windows, cygwin(windows)
#
#    get_terminal_size() - Return tuple (width, height) on
#                          linux, mac os x, windows,
#                          cygwin(windows)
#
#    _get_terminal_size_linux() -  Return tuple (width, height) on
#                          linux, mac os x and non-xterm on
#                          cygwin(windows)
#
#    _get_terminal_size_tput() - Return tuple (width, height) in
#                          cygwin's xterm on windows.
#
#    _get_terminal_size_windows() - Return tuple (width, height) on
#                          windows.
#
# Modifications:
#
#    2013/10/17 rsg Documented gist source and modifications.
#                   Converted print statement to print function.
#                   Added __header__ and current_os to print
#                   output.
#                   Created Python 3.x port from Python 2..x
#
# ToDo:
#
#    None
#
#################################################################

__title__     = 'tsGistGetTerminalSize'
__version__   = '1.0.0'
__date__      = '10/17/2013'
__authors__   = 'Richard S. Gordon'
__copyright__ = 'Copyright (c) 2013 ' + \
    '%s.\n\t\tAll rights reserved.' % __authors__
__license__   = 'GNU General Public License, ' + \
    'Version 3, 29 June 2007'
__credits__   = '\n\n  Credits: ' + \
                '\n\n\t  terminalsize (https://gist.github.com/' + \
                'jtriley/1108174) Features: ' + \
                '\n\t  Copyright (c) 2011 Justin T. Riley.' + \
                '\n\t\t\tAll rights reserved.' + \
                '\n\t  GNU General Public License, ' + \
                'Version 3, 29 June 2007'
 
__line1__ = '%s, v%s (build %s)' % (__title__, __version__, __date__)
__line2__ = 'Author(s): %s' % __authors__
__line3__ = '%s' % __copyright__

if ((len(__credits__) == 0)):
    __line4__ = '%s' % __license__
else:
    __line4__ = '%s%s' % (__license__, __credits__)

__header__ = '\n\n%s\n\n  %s\n  %s\n  %s\n' % (__line1__,
                                               __line2__,
                                               __line3__,
                                               __line4__)

mainTitleVersionDate = __line1__

#--------------------------------------------------------------------------

import os
import shlex
import struct
import platform
import subprocess

#--------------------------------------------------------------------------
 
def get_terminal_size():
    """ getTerminalSize()
     - get width and height of console
     - works on linux,os x,windows,cygwin(windows)
     originally retrieved from:
     http://stackoverflow.com/questions/566746/
            how-to-get-console-window-width-in-python
    """
    current_os = platform.system()
    tuple_xy = None
    if current_os == 'Windows':
        tuple_xy = _get_terminal_size_windows()
        if tuple_xy is None:
            tuple_xy = _get_terminal_size_tput()
            # needed for window's python in cygwin's xterm!
    if current_os in ['Linux', 'Darwin'] or current_os.startswith('CYGWIN'):
        tuple_xy = _get_terminal_size_linux()
    if tuple_xy is None:
        print("default")
        tuple_xy = (80, 25)      # default value
    return tuple_xy

#--------------------------------------------------------------------------
 
def _get_terminal_size_windows():
    try:
        from ctypes import windll, create_string_buffer
        # stdin handle is -10
        # stdout handle is -11
        # stderr handle is -12
        h = windll.kernel32.GetStdHandle(-12)
        csbi = create_string_buffer(22)
        res = windll.kernel32.GetConsoleScreenBufferInfo(h, csbi)
        if res:
            (bufx, bufy, curx, cury, wattr,
             left, top, right, bottom,
             maxx, maxy) = struct.unpack("hhhhHhhhhhh", csbi.raw)
            sizex = right - left + 1
            sizey = bottom - top + 1
            return sizex, sizey
    except:
        pass

#--------------------------------------------------------------------------

def _get_terminal_size_tput():
    # get terminal width
    # src: http://stackoverflow.com/questions/263890/
    #             how-do-i-find-the-width-height-of-a-terminal-window
    try:
        cols = int(subprocess.check_call(shlex.split('tput cols')))
        rows = int(subprocess.check_call(shlex.split('tput lines')))
        return (cols, rows)
    except:
        pass

#--------------------------------------------------------------------------
 
def _get_terminal_size_linux():
    def ioctl_GWINSZ(fd):
        try:
            import fcntl
            import termios
            cr = struct.unpack('hh',
                               fcntl.ioctl(fd, termios.TIOCGWINSZ, '1234'))
            return cr
        except:
            pass
    cr = ioctl_GWINSZ(0) or ioctl_GWINSZ(1) or ioctl_GWINSZ(2)
    if not cr:
        try:
            fd = os.open(os.ctermid(), os.O_RDONLY)
            cr = ioctl_GWINSZ(fd)
            os.close(fd)
        except:
            pass
    if not cr:
        try:
            cr = (os.environ['LINES'], os.environ['COLUMNS'])
        except:
            return None
    return int(cr[1]), int(cr[0])

#--------------------------------------------------------------------------
 
if __name__ == "__main__":

    print(__header__)

    sizex, sizey = get_terminal_size()
    print('\n  width = %d; height = %d' % (sizex, sizey))

    current_os = platform.system()
    print('\n  current_os = %s' % current_os)
